### DISCUSSION TITLE
Cursor pagination

### DISCUSSION THREAD

**Discussion by drmax24** [rebing/graphql-laravel#974] [Ideas]
Is there cursor pagination support? I thought there might be, but I can not see it anywhere... 

It is said to be the "best" pagination type in the official graphql docs
https://graphql.org/learn/pagination/

**Comment by mfn** [comment]
The built-in supported `PaginationType` is a `LengthAwarePaginator`, which I believe rules out cursor based pagination (usually you don't know / need to know how much there's to come automatically with the paginator component).

It's certainly possible to do it but there's nothing "out of the box" in this library. Cursor based pagination usually means you've a key / some keys which you take from your last element of your result and use it to construct the next page cursor value and then just return this with your payload.

Additionally, to avoid users playing around too much with those cursor keys, I encrypt them so they just can't change values as they like.
